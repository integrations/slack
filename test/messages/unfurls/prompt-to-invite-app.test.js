const PromptToInviteApp = require('../../../lib/messages/unfurls/prompt-to-invite-app');

describe('PromptToInviteApp message rendering', () => {
  test('works for public channels', async () => {
    const message = new PromptToInviteApp('C12345');
    expect(message.getAttachment()).toMatchSnapshot();
  });

  test('works for private channels', async () => {
    const message = new PromptToInviteApp('G12345');
    expect(message.getAttachment()).toMatchSnapshot();
  });

  test('works for direct messages', async () => {
    const message = new PromptToInviteApp('D12345');
    expect(message.getAttachment()).toMatchSnapshot();
  });
});
