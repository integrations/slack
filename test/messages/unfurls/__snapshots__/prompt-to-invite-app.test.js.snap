// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PromptToInviteApp message rendering works for direct messages 1`] = `
Object {
  "color": "#24292f",
  "fallback": "Invite @github to your channel",
  "mrkdwn_in": Array [
    "text",
  ],
  "text": "To receive these prompts in the channel where you shared the link, add GitHub to the conversation using /invite @github",
}
`;

exports[`PromptToInviteApp message rendering works for private channels 1`] = `
Object {
  "color": "#24292f",
  "fallback": "Invite @github to your channel",
  "mrkdwn_in": Array [
    "text",
  ],
  "text": "To receive these prompts in the channel where you shared the link, add GitHub to the conversation using /invite @github",
}
`;

exports[`PromptToInviteApp message rendering works for public channels 1`] = `
Object {
  "color": "#24292f",
  "fallback": "Invite @github to your channel",
  "mrkdwn_in": Array [
    "text",
  ],
  "text": "To receive these prompts in the channel where you shared the link (<#C12345> ), add GitHub to the conversation using /invite @github",
}
`;
